base_layer/core/src/base_node/chain_metadata_service/error.rs

```diff
@@ -21,20 +21,23 @@
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 use crate::base_node::comms_interface::CommsInterfaceError;
-use derive_error::Error;
 use prost::DecodeError;
 use tari_comms::message::MessageError;
 use tari_p2p::services::liveness::error::LivenessError;
+use thiserror::Error;
 
 #[derive(Debug, Error)]
 pub enum ChainMetadataSyncError {
-    /// Failed to decode chain metadata
-    DecodeError(DecodeError),
-    /// Peer did not send any chain metadata
+    #[error("Failed to decode chain metadata")]
+    DecodeError(#[from] DecodeError),
+    #[error("Peer did not send any chain metadata")]
     NoChainMetadata,
-    LivenessError(LivenessError),
-    CommsInterfaceError(CommsInterfaceError),
-    MessageError(MessageError),
-    /// Failed to publish `ChainMetadataEvent`
+    #[error("Liveness Error")]
```

Comment:
Few other places where the actual error is not being output.
 
```suggestion
    #[error("Liveness Error: {0}")]
```

Reply:
thanks @sdbondi - I thought that was only necessary for the "String" types, and #[from] took care of the others? 

How do I know which errors need that interpolation - is it all of them?

Reply:
`#[from]` generates a [`From`](https://doc.rust-lang.org/std/convert/trait.From.html) implementation to make it easy to convert say a `LivenessError` to a `ChainMetadataError`.

What `#[error("Liveness Error: {0}")]` is doing is calling the [`Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) implementation on `LivenessError` (which itself is implemented using `thiserror` and so has a `Display` implementation.

Reply:
So I think most of them should have the "inner error" output - basically it depends on the meaning of the error, which is hard to evaluate in bulk, but we definitely want to err on the side of displaying the info vs hiding it.